Added section about hourly weather forecast attributes

dannytsang opened the pull request:
## Proposed change
Also added example screenshot showing list structure of the forecast.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #26188

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

dannytsang committed:
Added section about hourly weather forecast attributes.

Added example screenshot showing list structure of the forecast.

frenck commented on source/_integrations/openweathermap.markdown:
```diff
@@ -104,3 +104,9 @@ The time period these sensors use depends on the forecast mode selected when con
+
+### Forecast Weather Conditions Intervals
+
+Periodic forecast information is stored in the `forecast` attribute on the entity. To access and use the information should be reserved for advanced users using [Templates](/docs/configuration/templating/).
```
This is a feature of the weather entity, should it be documented there?

dannytsang committed:
Moved hourly forecast information to weather integration.

Added reference to the weather integration on OpenWeatherMap so that people know it's an extension of the weather integration platform.

frenck commented on source/_integrations/openweathermap.markdown:
```diff
@@ -18,7 +18,7 @@ ha_platforms:
 ---
 
-The OpenWeatherMap weather integrations uses [OpenWeatherMap](https://openweathermap.org/) as a source for current meteorological data for your location.
+The OpenWeatherMap weather integrations uses [OpenWeatherMap](https://openweathermap.org/) as a source for current meteorological data for your location. It uses the [Weather integration platform](/integrations/weather/).
```
That is an odd sentence to write, we don't write such things anywhere. Additionally, it uses more entity components, not just the `weather` one.

```suggestion
The OpenWeatherMap weather integrations uses [OpenWeatherMap](https://openweathermap.org/) as a source for current meteorological data for your location.
```

dannytsang replied:
I will rethink this. I wanted a way to link back to the weather integration so people can see the common information like the forecast details.

frenck replied:
Inntegration can either list information on what they provide, some add chapters on their sensors and other entities.

A more general solution would be leveraging the `ha_platforms` frontend matter, to automatically list them in the aside maybe... (but would consider that out of scope for this PR).

frenck commented on source/_integrations/weather.markdown:
```diff
@@ -34,3 +34,9 @@ The `weather` platform only knows the below listed conditions. The reason for th
 - 'windy-variant'
 - 'exceptional'
+
+### Forecast Weather Conditions Intervals
```
```suggestion
## Forecast information
```

dannytsang replied:
Much better title. Updated in 3e25e8a57560e47829d5779f21f9806477a3674a

frenck commented on source/_integrations/weather.markdown:
```diff
@@ -34,3 +34,9 @@ The `weather` platform only knows the below listed conditions. The reason for th
+
+Periodic forecast information is stored in the `forecast` attribute on the entity. To access and use the information should be reserved for advanced users using [Templates](/docs/configuration/templating/).
+
+![Example OpenWeatherMap forecast attributes](/images/integrations/openweathermap/OpenWeatherMap_Attributes.png)
```
Can we inline a bit of a state YAML as an example instead? Screenshots are easy to get outdated.

dannytsang replied:
Good @frenck I will do that.

dannytsang replied:
Replaced image with YAML example. 3e25e8a57560e47829d5779f21f9806477a3674a

dannytsang committed:
Replaced screenshot with YAML forecast example.

Removed changes to OpenWeatherMap.

frenck reviewed (approved):
Thanks, @dannytsang 👍 

../Frenck
